File: workshop_content/KI_WORKSHOP/Java_Vergleichsgruppe/src/main/java/provisionierung/domain/Kategorie.java
package provisionierung.domain;

/**
 * Kundenkategorien mit ihren Basis-Provisionssätzen.
 */
public enum Kategorie {
    A(0.10),    // 10% Basisprovision
    B(0.07),    // 7% Basisprovision
    C(0.05),    // 5% Basisprovision
    SONSTIGE(0.01);  // 1% Basisprovision (Fallback)

    private final double basisSatz;

    Kategorie(double basisSatz) {
        this.basisSatz = basisSatz;
    }

    /**
     * Gibt den Basis-Provisionssatz für die Kategorie zurück.
     * @return Provisionsatz als Dezimalwert (z.B. 0.10 für 10%)
     */
    public double getBasisSatz() {
        return basisSatz;
    }
}

File: CSharp/Provisionierung/Domain/Kategorie.cs
namespace Provisionierung.Domain;

public enum Kategorie
{
    A,       // 10% Basisprovision
    B,       // 7%  Basisprovision
    C,       // 5%  Basisprovision
    Sonstige // 1%  Basisprovision (Fallback)
}

public static class KategorieExtensions
{
    public static decimal GetBasisSatz(this Kategorie kategorie) => kategorie switch
    {
        Kategorie.A => 0.10m,
        Kategorie.B => 0.07m,
        Kategorie.C => 0.05m,
        Kategorie.Sonstige => 0.01m,
        _ => throw new ArgumentOutOfRangeException(nameof(kategorie), $"Unbekannte Kategorie: {kategorie}")
    };
}
